crates/consensus/src/receipt/receipts.rs

```diff
@@ -102,6 +103,45 @@ impl<T> From<ReceiptWithBloom<T>> for Receipt<T> {
     }
 }
 
+/// A collection of receipts organized as a two-dimensional vector.
+#[derive(
+    Clone, Debug, PartialEq, Eq, Default, From, derive_more::Deref, DerefMut, IntoIterator,
+)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Receipts {
+    /// A two-dimensional vector of optional `Receipt` instances.
+    pub receipt_vec: Vec<Vec<Option<Receipt>>>,
```

Comment:
I don't want to do this, this is massive debt in reth because of bad pruning decisions

Reply:
Oh yeah absolutely agree, maybe this is a good time to review the reth design here when integrating this primitive, just changed the definition here